src/OrbitSsh/Credentials.cpp

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "OrbitSsh/Credentials.h"
+
+#include <absl/strings/str_join.h>
+
+#include <filesystem>
+
+#include "HomeEnvironmentVariable.h"
+#include "OrbitBase/File.h"
+#include "OrbitBase/Result.h"
+
+namespace orbit_ssh {
+
+ErrorMessageOr<std::filesystem::path> GetUsersSshConfigDirectory() {
+  const char* const home_directory_env_var = std::getenv(kHomeEnvironmentVariable);
+  if (home_directory_env_var == nullptr) {
+    return ErrorMessage{
+        "Could not determine the user's home directory location. Environment variable 'HOME' "
+        "doesn't exist."};
+  }
+
+  std::filesystem::path home_directory{home_directory_env_var};
+  OUTCOME_TRY(bool home_dir_exists, orbit_base::IsDirectory(home_directory));
+
+  if (!home_dir_exists) {
+    return ErrorMessage{absl::StrFormat(
+        "The user's home directory given by then 'HOME' environment variable does either not exist "
```

Comment:
Shouldn't you use "kHomeEnvironmentVariable" here instead of 'HOME' since it could be 'USERPROFILE'?. Also above.

Also, then --> the. 'does either not exist' --> 'does not exist'.

Reply:
Yes. Good catch. This is outdated. From before I added Windows support. Will fix...

Reply:
Done. Thx.